### PR TITLE
return resolved promise on version flag

### LINKED ISSUE #333: --version flag throws an error
Introduced in v0.13.0

Seems like a perfect storm of problems caused this:

1. Typescript didn't catch that undefined was being returned here, even when the implicit type is added. @justinfagnani is that a special behavior of the Promise type that it thinks it's in a `then()` style Promise chain?
2. Testing was covering the CLI directly, and not checking it's return type or the `/bin` file.

Good news is the version is still output, just with the additional error.

### PR DESCRIPTION
**Pull request by FredKSchott**
Fixes #333
Also adds implicit return type to run() function, plus tests.

/cc @justinfagnani

### CODE REVIEW

File: test/commands/cli_test.js [hunk 1]
```diff
@@ -25,58 +25,64 @@ suite('The general CLI', () => {
     let cli = new PolymerCli([]);
     let helpCommand = cli.commands.get('help');
     let helpCommandSpy = sinon.spy(helpCommand, 'run');
-    cli.run();
+    let cliRunResult = cli.run();
```
Comment (justinfagnani) [inline test/commands/cli_test.js]: this would be safer as:

```
return cli.run().then((_) => {
  assert.isOk(...);
});
```

Comment (FredKSchott) [inline test/commands/cli_test.js]: 👍  good call

**Comment by FredKSchott** [comment]
@justinfagnani PTAL

File: test/commands/cli_test.js [hunk 2]
```diff
@@ -17,67 +17,79 @@ const logging = require('plylog');
 const assert = require('chai').assert;
 const sinon = require('sinon');
 
-suite('The general CLI', () => {
+suite.only('The general CLI', () => {
 
   const defaultConfig = new Config();
 
-  test('displays general help when no command is called', () => {
+  test('displays general help when no command is called', (done) => {
```
Comment (justinfagnani) [inline test/commands/cli_test.js]: don't use `done`, just return a `Promise`

File: test/commands/cli_test.js [hunk 3]
```diff
@@ -17,67 +17,79 @@ const logging = require('plylog');
 const assert = require('chai').assert;
 const sinon = require('sinon');
 
-suite('The general CLI', () => {
+suite.only('The general CLI', () => {
```
Comment (justinfagnani) [inline test/commands/cli_test.js]: remove `.only`

Comment (FredKSchott) [inline test/commands/cli_test.js]: woof, good catch

File: test/commands/cli_test.js (hunk 2 above)
Comment (FredKSchott) [inline test/commands/cli_test.js]: huh, for some reason I thought mocha didn't support Promises natively. TIL

**Comment by FredKSchott** [comment]
@justinfagnani PTAL

File: src/polymer-cli.ts [hunk 4]
```diff
@@ -122,7 +122,7 @@ export class PolymerCli {
     return Array.from(mergedArgs.values());
   }
 
-  run() {
+  run(): Promise<any> {
```
Comment (rictic) [inline src/polymer-cli.ts]: When returning a promise we really want to use an async function to be sure that all returns and all errors are propagated consistently

Comment (justinfagnani) [inline src/polymer-cli.ts]: Agreed. @FredKSchott and I talked about doing a big conversion of cli and polymer-build to async/await

File: test/commands/cli_test.js [hunk 5]
```diff
@@ -25,59 +25,65 @@ suite('The general CLI', () => {
...
   });
 
   test('displays general help when no command is called with the --help flag', () => {
     let cli = new PolymerCli(['--help']);
     let helpCommand = cli.commands.get('help');
     let helpCommandSpy = sinon.spy(helpCommand, 'run');
-    cli.run();
-    assert.isOk(helpCommandSpy.calledOnce);
-    assert.deepEqual(helpCommandSpy.firstCall.args, [{command: null}, defaultConfig]);
+    cli.run().then(() => {
```
Comment (justinfagnani) [inline test/commands/cli_test.js]: `return cli.run().then(...`

File: src/polymer-cli.ts (hunk 4 above)
Comment (rictic) [inline src/polymer-cli.ts]: Yeah, and it's an easy thing to add onto a function that already returns a Promise. The only differences in that case are that with async:

  - a default return becomes `Promise.resolve();`
  - a throw in the stack frame of the function become a rejected promise

Comment (FredKSchott) [inline src/polymer-cli.ts]: 👍 I'm going to look into this for the entire repo next week

**Comment by justinfagnani** [comment]
LGTM

**Pull request merged**